pkg/api/store/ingress/store.go

```diff
@@ -59,17 +59,48 @@ func formatData(id string, data map[string]interface{}, forFrontend bool) {
 	}
 
 	// transform rules
-	if paths, ok := getPaths(data); ok {
+	if paths, ok, flag := getPaths(data); ok {
 		for hostpath, target := range paths {
 			updateRule(target, id, hostpath.host, hostpath.path, forFrontend, data, oldState, newState)
 		}
+		updateDataRules(data, flag)
 	}
 
 	updateCerts(data, forFrontend, oldState, newState)
 	setState(data, newState)
 	workload.SetPublicEnpointsFields(data)
 }
 
+func updateDataRules(data map[string]interface{}, flag bool) {
+	if !flag {
```

Comment:
would make sense to do the flag validation in the caller methods, and remove it from the signature as we return immediately if not set.